sdk/communication/azure-communication-phonenumbers/src/main/java/com/azure/communication/phonenumbers/PhoneNumbersAsyncClient.java

```diff
@@ -0,0 +1,377 @@
+     * @return {@link AcquiredPhoneNumber} representing the acquired telephone number.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<AcquiredPhoneNumber>> getPhoneNumberWithResponse(String phoneNumber) {
+        try {
+            Objects.requireNonNull(phoneNumber, "'phoneNumber' cannot be null.");
+            return client.getByNumberWithResponseAsync(phoneNumber);
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    /**
+     * Gets the list of the acquired phone numbers.
+     *
+     * @return A {@link PagedFlux} of {@link AcquiredPhoneNumber} instances representing acquired telephone numbers.
+     */
+    @ServiceMethod(returns = ReturnType.COLLECTION)
+    public PagedFlux<AcquiredPhoneNumber> listPhoneNumbers() {
+        try {
+            return client.listPhoneNumbersAsync(null, null);
+        } catch (RuntimeException ex) {
+            return new PagedFlux<>(() -> monoError(logger, ex));
+        }
+    }
+
+    /**
+     * Update an acquired phone number.
+     *
+     * @param phoneNumber The phone number id in E.164 format. The leading plus can be either + or encoded
+     *                    as %2B.
+     * @param update Update to an acquired phone number.
+     * @return A {@link Mono} containing
+     * a {@link AcquiredPhoneNumber} representing the acquired phone number
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<AcquiredPhoneNumber> updatePhoneNumber(String phoneNumber, PhoneNumberUpdateRequest update) {
+        try {
+            Objects.requireNonNull(phoneNumber, "'phoneNumber' cannot be null.");
+            Objects.requireNonNull(update, "'update' cannot be null.");
+            return client.updateAsync(phoneNumber, update);
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+
+    /**
+     * Update an acquired phone number with response.
+     *
+     * @param phoneNumber The phone number id in E.164 format. The leading plus can be either + or encoded
+     *                    as %2B.
+     * @param update Update to an acquired phone number.
+     * @return A {@link Mono} containing
+     * a {@link AcquiredPhoneNumber} representing the acquired phone number
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<AcquiredPhoneNumber>> 
+        updatePhoneNumberWithResponse(String phoneNumber, PhoneNumberUpdateRequest update) {
+        try {
+            Objects.requireNonNull(phoneNumber, "'phoneNumber' cannot be null.");
+            Objects.requireNonNull(update, "'update' cannot be null.");
+            return client.updateWithResponseAsync(phoneNumber, update);
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    /**
+     * Starts the search for available phone numbers to purchase.
+     *
+     * @param countryCode The ISO 3166-2 country code.
```

Comment:
May be we can add a link for ISO spec 'https://www.iso.org/standard/72483.html' if it helps developer.